wgsl/wgsl.recursive.bs.include

```diff
@@ -0,0 +1,693 @@
+
+ | [=right_syntax/attribute=] * `'override'` [=right_syntax/optionally_typed_ident=] ( [=syntax/equal=] [=right_syntax/expression=] )? [=syntax/semicolon=]
+
+ | [=right_syntax/attribute=] * `'var'` ( [=syntax/less_than=] [=right_syntax/address_space=] ( [=syntax/comma=] [=right_syntax/access_mode=] )? [=syntax/greater_than=] )? [=right_syntax/optionally_typed_ident=] ( [=syntax/equal=] [=right_syntax/expression=] )? [=syntax/semicolon=]
+
+ | [=syntax/semicolon=]
+
+ | `'static_assert'` [=right_syntax/expression=] [=syntax/semicolon=]
+
+ | `'struct'` [=right_syntax/ident=] [=syntax/brace_left=] [=right_syntax/attribute=] * [=right_syntax/member_ident=] [=syntax/colon=] [=right_syntax/type_decl=] ( [=syntax/comma=] [=right_syntax/attribute=] * [=right_syntax/member_ident=] [=syntax/colon=] [=right_syntax/type_decl=] )* [=syntax/comma=] ? [=syntax/brace_right=]
+
+ | `'type'` [=right_syntax/ident=] [=syntax/equal=] [=right_syntax/type_decl=] [=syntax/semicolon=]
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>hex_float_literal</dfn>:
+
+ | `/0[xX][0-9a-fA-F]*\.[0-9a-fA-F]+([pP][+-]?[0-9]+[fh]?)?/`
+
+ | `/0[xX][0-9a-fA-F]+[pP][+-]?[0-9]+[fh]?/`
+
+ | `/0[xX][0-9a-fA-F]+\.[0-9a-fA-F]*([pP][+-]?[0-9]+[fh]?)?/`
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>ident</dfn>:[=syntax/ident_pattern_token=]
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>int_literal</dfn>:
+
+ | `/0[iu]?/`
+
+ | `/0[xX][0-9a-fA-F]+[iu]?/`
+
+ | `/[1-9][0-9]*[iu]?/`
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>interpolation_sample_name</dfn>:
+
+ | `'center'`
+
+ | `'centroid'`
+
+ | `'sample'`
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>interpolation_type_name</dfn>:
+
+ | `'flat'`
+
+ | `'linear'`
+
+ | `'perspective'`
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>language</dfn>:
+
+ | [=right_syntax/translation_unit=] `EndOfText`
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>lhs_expression</dfn>:
+
+ | ( [=syntax/star=] | [=syntax/and=] ) * [=right_syntax/core_lhs_expression=] [=right_syntax/postfix_expression=] ?
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>mat_prefix</dfn>:
+
+ | `'mat2x2'`
+
+ | `'mat2x3'`
+
+ | `'mat2x4'`
+
+ | `'mat3x2'`
+
+ | `'mat3x3'`
+
+ | `'mat3x4'`
+
+ | `'mat4x2'`
+
+ | `'mat4x3'`
+
+ | `'mat4x4'`
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>member_ident</dfn>:[=syntax/ident_pattern_token=]
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for='right_syntax'>multiplicative_expression</dfn>:
+
+ | [=right_syntax/multiplicative_expression=]
```

Comment:
Something has gone wrong here ....

Reply:
yep.  The left-factoring code was rewriting the original rule, then immediately reabsorbing. 
Fixed now.